huaweicloud/services/er/resource_huaweicloud_er_flow_log.go

```diff
@@ -0,0 +1,412 @@
+	getFlowLogPath = strings.ReplaceAll(getFlowLogPath, "{er_id}", d.Get("instance_id").(string))
+	getFlowLogPath = strings.ReplaceAll(getFlowLogPath, "{flow_log_id}", d.Id())
+	getFlowLogOpts := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+		OkCodes: []int{
+			200,
+		},
+	}
+	resp, err := client.Request("GET", getFlowLogPath, &getFlowLogOpts)
+	if err != nil {
+		return nil, err
+	}
+
+	return resp, nil
+}
+
+func resourceFlowLogRead(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	getFlowLogResp, err := getFlowLogInfo(d, meta)
+	if err != nil {
+		return common.CheckDeletedDiag(d, err, "ER flow log")
+	}
+
+	getFlowLogRespBody, err := utils.FlattenResponse(getFlowLogResp)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+	mErr := multierror.Append(
+		nil,
+		d.Set("region", region),
+		d.Set("log_store_type", utils.PathSearch("flow_log.log_store_type", getFlowLogRespBody, nil)),
+		d.Set("log_group_id", utils.PathSearch("flow_log.log_group_id", getFlowLogRespBody, nil)),
+		d.Set("log_stream_id", utils.PathSearch("flow_log.log_stream_id", getFlowLogRespBody, nil)),
+		d.Set("name", utils.PathSearch("flow_log.name", getFlowLogRespBody, nil)),
+		d.Set("description", utils.PathSearch("flow_log.description", getFlowLogRespBody, nil)),
+		d.Set("resource_type", utils.PathSearch("flow_log.resource_type", getFlowLogRespBody, nil)),
+		d.Set("resource_id", utils.PathSearch("flow_log.resource_id", getFlowLogRespBody, nil)),
+		d.Set("state", utils.PathSearch("flow_log.state", getFlowLogRespBody, nil)),
+		d.Set("created_at", utils.PathSearch("flow_log.created_at", getFlowLogRespBody, nil)),
+		d.Set("updated_at", utils.PathSearch("flow_log.updated_at", getFlowLogRespBody, nil)),
+		d.Set("enabled", utils.PathSearch("flow_log.enabled", getFlowLogRespBody, nil)),
+	)
+
+	return diag.FromErr(mErr.ErrorOrNil())
+}
+
+func updateFlowLogState(client *golangsdk.ServiceClient, d *schema.ResourceData, operateType string) error {
+	updateFlowLogStateHttpUrl := "enterprise-router/{er_id}/flow-logs/{flow_log_id}/" + operateType
+	updateFlowLogStatePath := client.ResourceBaseURL() + updateFlowLogStateHttpUrl
+	updateFlowLogStatePath = strings.ReplaceAll(updateFlowLogStatePath, "{er_id}", d.Get("instance_id").(string))
+	updateFlowLogStatePath = strings.ReplaceAll(updateFlowLogStatePath, "{flow_log_id}", d.Id())
+
+	updateFlowLogStateOpts := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+		OkCodes: []int{
+			202,
+		},
+	}
+	_, err := client.Request("POST", updateFlowLogStatePath, &updateFlowLogStateOpts)
+	if err != nil {
+		return fmt.Errorf("error %s flow log: %s", operateType, err)
+	}
+
+	return nil
+}
+
+func buildUpdateFlowLogBodyParams(d *schema.ResourceData) map[string]interface{} {
+	bodyParams := map[string]interface{}{
+		"name":        utils.ValueIngoreEmpty(d.Get("name")),
+		"description": utils.ValueIngoreEmpty(d.Get("description")),
```

Comment:
If the description can be updated with empty value, we shouldn't ingore the value.

Reply:
fixed